### PR TITLE
Nate/design fix/LOOP-1419/copy color update

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1419

Also see: https://github.com/tidepool-org/LoopKit/pull/136

Note the refactor coming with this update. The bluetooth status highlights in the pills are coordinated messaging, meaning the left pill presents one part and the right pill presents that other part. This requires a class with enough visibility to coordinate this effort. In this case, I chose the `DeviceDataManager`.

Since the `DeviceDataManager` is now responsible for some of the device status highlights, I've made it responsible for all of the device status highlight, tap actions, and device lifecycle progress. 

Also note that I overlooked tap actions that break DIY (pump HUD provider tap and cgm app URL). These are fixed with this update.

### CODE REVIEW

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -1307,40 +1280,33 @@ final class StatusTableViewController: ChartsTableViewController {
...
         }
-    }
-    
-    @objc private func hudViewTapped(_ sender: UIGestureRecognizer) {
-        if let hudSubView = sender.view as? BaseHUDView,
-            let pumpManagerHUDProvider = deviceManager.pumpManagerHUDProvider,
-            let action = pumpManagerHUDProvider.didTapOnHUDView(hudSubView)
-        {
+        
+        if let action = action {
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift]: [nit] Consider
```
guard let action = action else {
    return
}
```
here?

File: Loop/View Controllers/StatusTableViewController.swift [hunk 2]
```diff
@@ -1243,10 +1220,10 @@ final class StatusTableViewController: ChartsTableViewController {
             hudView.loopCompletionHUD.addGestureRecognizer(statusTapGestureRecognizer)
             hudView.loopCompletionHUD.accessibilityHint = NSLocalizedString("Shows last loop error", comment: "Loop Completion HUD accessibility hint")
             
-            let pumpStatusTapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(pumpStatusTapped(_:)))
+            let pumpStatusTapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(deviceStatusTapped(_:)))
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift]: [nit] Why get rid of the separate `Tapped` handlers for CGM and pump when down below in `deviceStatusTapped` you just have to figure out whether it was CGM or Pump again (e.g. via `sender.view as? CGMStatusHUDView`)?

File: Loop/Managers/DeviceDataManager.swift [hunk 3]
```diff
@@ -866,3 +870,13 @@ extension DeviceDataManager {
         }
     }
 }
+
+//MARK: - Bluetooth State Manager Obversation
+
+extension DeviceDataManager: BluetoothStateManagerObserver {
+    func bluetoothStateManager(_ bluetoothStateManager: BluetoothStateManager,
+                           bluetoothStateDidUpdate bluetoothState: BluetoothStateManager.BluetoothState)
```
Comment (darinkrauss) [inline Loop/Managers/DeviceDataManager.swift]: [nit] Indentation.

File: Loop/Extensions/DeviceDataManager+DeviceStatus.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            return (cgmManager as? CGMManagerUI)?.cgmStatusHighlight
+        }
+    }
+    
+    var cgmLifecycleProgress: DeviceLifecycleProgress? {
+        return (cgmManager as? CGMManagerUI)?.cgmLifecycleProgress
+    }
+    
+    var pumpStatusHighlight: DeviceStatusHighlight? {
+        if bluetoothState == .denied ||
```
Comment (darinkrauss) [inline Loop/Extensions/DeviceDataManager+DeviceStatus.swift:32]: [nit] Why are the Bluetooth conditions and highlight different between CGM and pump? Seems like they should be the same.

File: Loop/Extensions/DeviceDataManager+DeviceStatus.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        return AddDeviceStatusHighlight(localizedMessage: NSLocalizedString("Add Pump", comment: "Title text for button to set up a Pump"))
+    }
+    
+    struct AddDeviceStatusHighlight: DeviceStatusHighlight {
+        var localizedMessage: String
+        var imageSystemName: String = "plus.circle"
+        var state: DeviceStatusHighlightState = .normal
+    }
+    
+    func didTapOnPumpStatus(_ view: BaseHUDView? = nil) -> HUDTapAction? {
```
Comment (darinkrauss) [inline Loop/Extensions/DeviceDataManager+DeviceStatus.swift]: [nit] Really minor nit, but consider ordering of CGM and Pump functionality in the same order through out file. Above you have CGM before Pump, but here you have Pump then CGM.

**Review by darinkrauss — APPROVED** [review]
A few nits and questions, but otherwise LGTM! 💺

File: Loop/View Controllers/StatusTableViewController.swift (hunk 2 above)
Comment (nhamming) [reply Loop/View Controllers/StatusTableViewController.swift]: I wanted to reduce code, but thinking about it more, I'll separate the execution of the action from the gathering of the action.

File: Loop/Extensions/DeviceDataManager+DeviceStatus.swift (hunk 4 above)
Comment (nhamming) [reply Loop/Extensions/DeviceDataManager+DeviceStatus.swift:32]: Yes, I thought they should be as well, but design requires different bluetooth highlights.

**Pull request merged**